src/_P097_Esp32Touch.ino

```diff
@@ -150,25 +197,30 @@ boolean Plugin_097(uint8_t function, struct EventStruct *event, String& string)
             if (touched) {
               if (P097_SEND_TOUCH_EVENT) {
                 // schedule a read to update output values and send to controllers
+                UserVar.setFloat(event->TaskIndex, 1, 1);
                 Scheduler.schedule_task_device_timer(event->TaskIndex, millis());
+                
               }
               bitSet(p097_pinTouchedPrev, t);
-            } else {
+            } else { 
+              // Touch released
               if (P097_SEND_RELEASE_EVENT) {
                 // schedule a read to update output values and send to controllers
+                UserVar.setFloat(event->TaskIndex, 1, 0);
                 Scheduler.schedule_task_device_timer(event->TaskIndex, millis());
+                
               }
 
               if (P097_SEND_DURATION_EVENT) {
-                if (Settings.UseRules) {
                   eventQueue.add(event->TaskIndex, F("Duration"), timePassedSince(p097_timestamp[t]));
-                }
```

Comment:
This check is there for a reason...

Reply:
Ohh... i missed that. I was working the whole day on this little piece of code mostly trying to add long-press 🥴
But.... i see now, that i could add a toggle option :)